dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessService.java

```diff
@@ -2055,4 +2111,199 @@ public String changeJson(ProcessData processData, String oldJson) {
+            // TODO update by code directly
+            TaskDefinition taskDefinition = taskDefinitionMapper.queryByDefinitionName(project.getCode(), task.getName());
+            updateTaskDefinition(operator, project.getCode(), task, taskDefinition);
+        }
+        createTaskAndRelation(operator, project.getName(), "", processDefinition, processData);
+        ProcessDefinitionLog processDefinitionLog = updateProcessLog(operator, processDefinition.getCode(),
+                name, processData, project, desc, locations, connects);
+        return switchVersion(processDefinition, processDefinitionLog);
+    }
+
+    /**
+     * @param operator
+     * @param processDefinitionCode
+     * @param processDefinitionName
+     * @param processData
+     * @param project
+     * @param desc
+     * @param locations
+     * @param connects
+     * @return
+     */
+    public ProcessDefinitionLog updateProcessLog(User operator, Long processDefinitionCode, String processDefinitionName,
+                                                 ProcessData processData, Project project,
+                                                 String desc, String locations, String connects) {
+        ProcessDefinitionLog processDefinitionLog = new ProcessDefinitionLog();
+        int version = processDefinitionLogMapper.queryMaxVersionForDefinition(processDefinitionLog.getCode());
+        processDefinitionLog.setCode(processDefinitionCode);
+        processDefinitionLog.setVersion(version);
+        processDefinitionLog.setName(processDefinitionName);
+        processDefinitionLog.setReleaseState(ReleaseState.OFFLINE);
+        processDefinitionLog.setProjectCode(project.getCode());
+        processDefinitionLog.setDescription(desc);
+        processDefinitionLog.setLocations(locations);
+        processDefinitionLog.setConnects(connects);
+        processDefinitionLog.setTimeout(processData.getTimeout());
+        processDefinitionLog.setTenantId(processData.getTenantId());
+        processDefinitionLog.setOperator(operator.getId());
+        Date updateTime = new Date();
+        processDefinitionLog.setOperateTime(updateTime);
+
+        //custom global params
+        List<Property> globalParamsList = new ArrayList<>();
+        if (CollectionUtils.isNotEmpty(processData.getGlobalParams())) {
+            Set<Property> userDefParamsSet = new HashSet<>(processData.getGlobalParams());
+            globalParamsList = new ArrayList<>(userDefParamsSet);
+        }
+        processDefinitionLog.setGlobalParamList(globalParamsList);
+        processDefinitionLog.setUpdateTime(updateTime);
+        processDefinitionLog.setFlag(Flag.YES);
+
+        processDefinitionLog.setOperator(operator.getId());
+        processDefinitionLog.setOperateTime(processDefinitionLog.getUpdateTime());
+        int insert = processDefinitionLogMapper.insert(processDefinitionLog);
+        if (insert > 0) {
+            return processDefinitionLog;
+        }
+        return null;
```

Comment:
This 'return null',and no judgment in switchversion, it may be happened NullPointerException

Reply:
"null judgement" exists in "switchVersion"
may be you miss it.

Reply:
"null judgement" exists in "switchVersion"
may be you miss it.